modules/battery-udev.c

```diff
@@ -1720,6 +1720,15 @@ udevdevice_evaluate_battery(udevdevice_t *self, mcebat_t *mcebat)
         mcebat->battery_status = BATTERY_STATUS_OK;
         if( self->udd_full && capacity >= BATTERY_CAPACITY_FULL )
             mcebat->battery_status = BATTERY_STATUS_FULL;
+        /* Some devices never report Full state */
+        else if ( capacity >= 100 ) {
+            mcebat->battery_status = BATTERY_STATUS_FULL;
```

Comment:
This ought to be already taken care by "charging -> not-charging transition while capacity >= limit" heuristics few lines below... or do you mean that there is some device that gets stuck in "charging" state forever?

(Not getting battery full immediately after reaching 100% is more or less expected and change like this would trigger premature battery full signaling on well behaving devices).

Reply:
Yes, my X10 II seems to stuck forever in Charging state. It was at 100% capacity for longer time and I never saw Full under sysfs. And since the code below is in an `else if` block, it never reaches it.

Reply:
omg. yet another kind of potential weird behavior...  As I have x10 ii as daily device and I do not recall seeing such thing: how was your charging setup? pc/wall charger? display on/off? live tracing over wlan? other sources of battery drain? i.e. anything abnormal that would/could affect final stages of charging. 

It would be nice to not have weird scenarios affecting well behaving devices, but maybe this kind of kiss solution is okay in the context of this PR.

Reply:
Originally tested with cable connected to laptop, later using a wall charger, both behaving equally WRT this bug. Both with display on and off, monitoring over wlan but also checking journal afterwards. It took a while to determine what actually happens there so I could exercise various options :)

Reply:
Basically IMO: if you just connect a charger/pc, allow display to blank, leave the device to charge without doing anything special and it reaches battery full in somewhat sane proximity to 100% (led pattern / after the fact journal inspection) -> we should not add heuristics to second guess kernel reporting.